src/plots/piePlot.ts

```diff
@@ -243,6 +267,73 @@ export module Plots {
       var avgAngle = (startAngle + endAngle) / 2;
       return { x: avgRadius * Math.sin(avgAngle), y: -avgRadius * Math.cos(avgAngle) };
     }
+
+    protected _additionalPaint(time: number) {
+      if (this._labelsEnabled) {
+        Utils.Window.setTimeout(() => this._drawLabels(), time);
+      }
+    }
+
+    private _drawLabels() {
+      var attrToProjector = this._generateAttrToProjector();
+      var labelArea = this._renderArea.append("g").classed("label-area", true);
+      var measurer = new SVGTypewriter.Measurers.Measurer(labelArea);
+      var writer = new SVGTypewriter.Writers.Writer(measurer);
+      this.entities().forEach((entity) => {
+        var value = "" + this.sectorValue().accessor(entity.datum, entity.index, entity.dataset);
+        var measurement = measurer.measure(value);
+
+        var theta = (this._endAngles[entity.index] + this._startAngles[entity.index]) / 2;
+        var outerRadius = this.outerRadius().accessor(entity.datum, entity.index, entity.dataset);
+        if (this.outerRadius().scale) {
+          outerRadius = this.outerRadius().scale.scale(outerRadius);
+        }
+        var innerRadius = this.innerRadius().accessor(entity.datum, entity.index, entity.dataset);
+        if (this.innerRadius().scale) {
+          innerRadius = this.innerRadius().scale.scale(innerRadius);
+        }
+        var labelRadius = (outerRadius + innerRadius) / 2;
+
+        var x = Math.sin(theta) * labelRadius - measurement.width / 2;
+        var y = -Math.cos(theta) * labelRadius - measurement.height / 2;
+
+        // Hide the label if it is outside of the slice area
+        var svgX = x + this.width() / 2;
+        var svgY = y + this.height() / 2;
+        var corners = [
+          { x: svgX, y: svgY },
+          { x: svgX + measurement.width, y: svgY },
+          { x: svgX + measurement.width, y: svgY + measurement.height },
+          { x: svgX, y: svgY + measurement.height}
+        ];
+
+        var showLabel = true;
+        corners.forEach((corner) => {
+          var entities = this.entitiesAt(corner);
+          if (entities.length < 1) {
+            showLabel = false;
+          } else {
+            if (entities[0].index != entity.index) {
```

Comment:
@jtlan I'm curious if there's a good way to do object equality between `Entities`. This works because there's only a single Entity/Dataset to ever be concerned with in this plot but this is definitely not generalizable.